### PR TITLE
added listner to RemoteDeliveryCreatedEvent

### PR DESCRIPTION
**Pull request by mike-ionut-mihai-sandu-tao**
DataStore][BE] Gather metadata information and parse to storage format

Ticket: https://oat-sa.atlassian.net/browse/CON-288

this will generate 3 files :
image

the files values are jsonLD representation of the metaData for each resource 
 - delivery
 - test
 - items(all of them)

the Task will use GCP file-system driver and store them inside the bucket

### CODE REVIEW

File: manifest.php [hunk 1]
```diff
@@ -15,8 +15,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- * Copyright (c) 2016 (original work) Open Assessment Technologies SA;
- *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
```
Comment (bartlomiejmarszal) [inline manifest.php:18]:
```suggestion
 * Copyright (c) 2016-2021 (original work) Open Assessment Technologies SA;
```

File: migrations/Version202101281349443635_taoPublishing.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    public function getDescription(): string
+    {
+        return 'Registering listener for RemoteDeliveryCreatedEvent';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $eventManager = $this->getEventManger();
+        $eventManager->attach(
```
Comment (bartlomiejmarszal) [inline migrations/Version202101281349443635_taoPublishing.php:45]: I think you need to reregister event manager with new event attached.

File: migrations/Version202101281349443635_taoPublishing.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            DeliveryCreatedEvent::class,
+            [DeliveryMetadataListener::class, 'whenDeliveryIsPublished']
+        );
+    }
+
+    public function down(Schema $schema): void
+    {
+        $eventManager = $this->getEventManger();
+        $eventManager->detach(
```
Comment (bartlomiejmarszal) [inline migrations/Version202101281349443635_taoPublishing.php:54]: I think you need to reregister event manager with new event attached.

**Review by bartlomiejmarszal — CHANGES_REQUESTED** [review]
You need to reregister event manager with new event attached.

**Review by bartlomiejmarszal — COMMENTED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: scripts/install/RegisterDataStoreServices.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+use oat\oatbox\extension\InstallAction;
+use oat\taoDeliveryRdf\model\DataStore\DeliveryMetadataListener;
+use oat\taoPublishing\model\publishing\event\RemoteDeliveryCreatedEvent;
+
+class RegisterDataStoreServices extends InstallAction
+{
+
+    public function __invoke($params)
+    {
+        $eventManager = $this->getEventManger();
```
Comment (gabrielfs7) [inline scripts/install/RegisterDataStoreServices.php:35]: Here too https://github.com/oat-sa/extension-tao-publishing/pull/94/files#r566724640

File: manifest.php [hunk 5]
```diff
@@ -29,26 +28,26 @@
...
-        array('grant', 'http://www.tao.lu/Ontologies/generis.rdf#taoPublishingManager', array('ext'=>'taoPublishing')),
+        array('grant', 'http://www.tao.lu/Ontologies/generis.rdf#taoPublishingManager', array('ext' => 'taoPublishing')),
     ),
     'install' => array(
         'rdf' => array(
-            __DIR__. '/model/ontology/platform.rdf',
-            __DIR__. '/model/ontology/taodelivery.rdf',
-            __DIR__. '/model/ontology/taotest.rdf',
-            __DIR__. '/model/ontology/indexation.rdf'
+            __DIR__ . '/model/ontology/platform.rdf',
```
Comment (gabrielfs7) [inline manifest.php:47]: Missing add new installation script call

File: migrations/Version202101281349443635_taoPublishing.php (hunk 3 above)
Comment (mike-ionut-mihai-sandu-tao) [reply migrations/Version202101281349443635_taoPublishing.php:55]: thank you, i will commit this part

File: manifest.php (hunk 5 above)
Comment (mike-ionut-mihai-sandu-tao) [reply manifest.php:48]: added thank you

File: migrations/Version202101281349443635_taoPublishing.php (hunk 2 above)
Comment (mike-ionut-mihai-sandu-tao) [reply migrations/Version202101281349443635_taoPublishing.php:45]: added thank you for the heads up

File: scripts/install/RegisterDataStoreServices.php (hunk 4 above)
Comment (mike-ionut-mihai-sandu-tao) [reply scripts/install/RegisterDataStoreServices.php:35]: done thank you

File: manifest.php (hunk 5 above)
Comment (mike-ionut-mihai-sandu-tao) [reply manifest.php:48]: done

File: scripts/install/RegisterDataStoreServices.php (hunk 4 above)
Comment (mike-ionut-mihai-sandu-tao) [reply scripts/install/RegisterDataStoreServices.php:35]: done

**Review by gabrielfs7 — APPROVED** [review]
LGTM:

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by shpran — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by boajer — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**